File: frontend2/RentingServiceFrontend/src/app/interfaces/reservationDto.ts
import { postDto } from './postDto';

export interface reservationDto {
  reservationId: number;
  userId: number;
  postId: number;
  fromDate: Date;
  toDate: Date;
  reservationFlag: ReservationFlagEnum;
  post: postDto;
  price: number;
}

export enum ReservationFlagEnum {
  'Created' = 0,
  'Confirmed' = 1,
  'Cancelled' = 2,
  'Completed' = 3,
}

File: backend/RentingServiceBackend/Models/ReservationDto.cs
using RentingServiceBackend.Entities;

namespace RentingServiceBackend.Models
{
    public class ReservationDto
    {
        public int ReservationId { get; set; }
        public int UserId { get; set; }
        public int PostId { get; set; }
        public DateTime FromDate { get; set; }
        public DateTime ToDate { get; set; }
        public ReservationFlagEnum ReservationFlag { get; set; }
        public PostDto Post { get; set; }
        public double Price {  get; set; }
    }
}
